### PR TITLE
Improved documentation

### PR DESCRIPTION
**Pull request by dmarrazzo**
new project structure section, Domain Model introduction and constraints rules description.

### CODE REVIEW

**Comment by kie-ci5** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /target
 /local
 
+# hidden dir
+.*
```
Comment (Christopher-Chianelli) [inline .gitignore]: I think that ".\*" might have some unintended effects; probably best to make it mirror Optaplanner .gitignore: https://github.com/kiegroup/optaplanner/blob/master/.gitignore (which uses "/.*" instead (i.e. only ignore hidden files at root))

**Comment by Christopher-Chianelli** [comment]
ok to test

**Comment by dmarrazzo** [comment]
@Christopher-Chianelli I fixed the .gitignore.
I believe that it's ready to be merged
Thank you

**Comment by Christopher-Chianelli** [comment]
@dmarrazzo please rebase this branch on master; I have already updated the .gitignore in a previous commit (https://github.com/kiegroup/optaweb-employee-rostering/blob/master/.gitignore).

**Comment by dmarrazzo** [comment]
@Christopher-Chianelli rebased and squashed. Thank you

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,10 @@
 /target
 /local
 
+# hidden dir
```
Comment (Christopher-Chianelli) [inline .gitignore]: This is already handled by the lines below it. Please remove these changes.

**Comment by dmarrazzo** [comment]
@Christopher-Chianelli ok sorry for all this recycle, I'm quite noob of contribution.
Now I believe that works.

**Comment by Christopher-Chianelli** [comment]
jenkins retest this

**Comment by Christopher-Chianelli** [comment]
@dmarrazzo Thanks for the changes! Don't worry, stuff like this happens when two PR's incorporate the same changes. It always best to rebase your branches on master whenever master changes. The build worked on my machine, but it appears Jenkins might have an issue.

**Comment by Christopher-Chianelli** [comment]
jenkins retest this

**Comment by Christopher-Chianelli** [comment]
jenkins retest this

**Comment by ge0ffrey** [comment]
@cuijulian Can you take a look at this and work through any merge conflicts and changes you deem needed to get this merged? (cc  @Christopher-Chianelli )

**Comment by Christopher-Chianelli** [comment]
Will do; Jenkins was always failing before, which is why this wasn't merged eariler

**Comment by cuijulian** [comment]
@ge0ffrey Sure, I will create a separate PR that includes the changes in this PR that are still relevant after the refactor (if there are any)

@dmarrazzo Thank you for the PR. For context, OptaWeb Employee Rostering was refactored from gwt/Wildfly to Reactjs/Spring Boot, so a lot of the documentation is now obsolete.

**Comment by triceo** [comment]
`optaweb-employee-rostering` project is no longer maintained, closing this PR.

**Pull request closed without merging**